src/net_builtin.c

```diff
@@ -654,16 +665,17 @@ static void read_conn(struct mg_connection *c, struct pkt *pkt) {
     MG_VERBOSE(("%lu SEQ %x -> %x", c->id, mg_htonl(pkt->tcp->seq), s->ack));
     // Advance ACK counter
     s->ack = (uint32_t) (mg_htonl(pkt->tcp->seq) + pkt->pay.len);
-    s->noack += pkt->pay.len;
-    if (s->noack + 1500 >= MIP_TCP_WIN) {
+    s->unacked += pkt->pay.len;
+    // size_t diff = s->acked <= s->ack ? s->ack - s->acked : s->ack;
+    if (s->unacked > MIP_TCP_WIN / 2 && s->acked != s->ack) {
```

Comment:
I think this right side check is needed because we don't zero `->unacked` when we ACK on timeout.

My belief is that `unacked` will slowly grow and reach `MIP_TCP_WIN / 2` while ACKs are being sent on timeouts (usual traffic) as we've been doing. Then next frame received will trigger an unnecessary immediate ACK. Nothing to worry about, though.

I still fail to see the mechanism for those many dup acks in a row, so let's not make changes here yet.